controllers/argocd/policyrule.go

```diff
@@ -229,6 +240,17 @@ func policyRuleForServerApplicationSourceNamespaces() []v1.PolicyRule {
 				"delete",
 			},
 		},
+		{
+			APIGroups: []string{
+				"batch",
+			},
+			Resources: []string{
+				"jobs",
+			},
+			Verbs: []string{
+				"create",
+			},
+		},
```

Comment:
@chengfang this block is not required here. This role is specifically meant only to grant permissions to the server  to manipulate applications in non-control plane namespaces